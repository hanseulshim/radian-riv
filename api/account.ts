import Cookies from 'js-cookie'
import { handleApi } from './index'

interface Profile {
  name_first: string
  name_last: string
  title: string
  address: string
  city: string
  state: string
  zip: string
  department: string
  phone_mobile: string
  phone_home: string
}

export const submitProfile = async (form: Profile): Promise<string> => {
  const data = await handleApi('/user/update', form)
  const authCookies = Cookies.get('auth')
  const auth = JSON.parse(authCookies)
  Cookies.set('auth', { ...auth, user: { ...form } })
  return data
}

interface ChangePassword {
  userid_ssid: string
  pwd: string
}

export const submitChangePassword = async (
  form: ChangePassword
): Promise<string> => {
  const { pwd } = form
  if (pwd.length < 8) {
    throw new Error('Password must be at least 8 characters(s) long')
  } else if (!/[A-Z]/.test(pwd)) {
    throw new Error('Password must contain an uppercase character')
  } else if (!/[a-z]/.test(pwd)) {
    throw new Error('Password must contain a lowercase character')
  } else if (!/[0-9]/.test(pwd)) {
    throw new Error('Password must contain a numeric character')
  } else if (!/[!@#$]/.test(pwd)) {
    throw new Error(
      'Password must contain a special character (example: !,@,#,$)'
    )
  }
  const data = await handleApi('/auth/pwdchange', form)
  return data
}

interface Question {
  questionid: number
  question_text: string
}
export const getSecurityQuestions = async (): Promise<Question[]> => {
  const questions = await handleApi('/auth/questions')
  return questions
}

interface SecurityQuestions {
  userid_ssid: string
  question1id: number
  answer1: string
  question2id: number
  answer2: string
  question3id: number
  answer3: string
}

export const setSecurityQuestions = async (
  form: SecurityQuestions
): Promise<string> => {
  if (
    form.question1id === null ||
    form.question2id === null ||
    form.question3id === null
  ) {
    throw new Error(`Questions can't be blank`)
  }
  if (form.answer1 === 'error') {
    throw new Error('Error with security question')
  }
  const data = await handleApi('/auth/questionsset', form)
  return data
}

interface FilterDefaultOption {
  label: string
  value: string | number
}

interface FilterDefaults {
  sqft?: FilterDefaultOption
  sqft_min?: number
  sqft_max?: number
  sqft_percent?: FilterDefaultOption
  comparable_retail: boolean
  comparable_distressed: boolean
  time_going_back: FilterDefaultOption
  comps_subdivision: boolean
  restrict_comps: FilterDefaultOption
}

interface FilterDefaultSet {
  sqft?: number
  sqft_min?: number
  sqft_max?: number
  sqft_percent?: number
  comparable_retail: boolean
  comparable_distressed: boolean
  time_going_back: number
  comps_subdivision: boolean
  restrict_comps: number
}

export const getFilterDefaults = async (
  userid: string
): Promise<FilterDefaults> => {
  const defaults = await handleApi(`/user/filter-defaults/${userid}`)
  return defaults
}

export const setFilterDefaults = async (
  form: FilterDefaultSet
): Promise<string> => {
  const data = await handleApi('/user/filter-defaults-set', form)
  return data
}

export const getFilterDefaultsSquareFt = async (
  userid: string
): Promise<FilterDefaultOption[]> => {
  const data = await handleApi(`/user/filter-defaults-sqft/${userid}`)
  return data
}

export const getFilterDefaultsSquareFtPercent = async (
  userid: string
): Promise<FilterDefaultOption[]> => {
  const data = await handleApi(`/user/filter-defaults-sqft-percent/${userid}`)
  return data
}

export const getFilterDefaultsRestrict = async (
  userid: string
): Promise<FilterDefaultOption[]> => {
  const data = await handleApi(`/user/filter-defaults-restrict-comps/${userid}`)
  return data
}

export const getFilterDefaultsTime = async (
  userid: string
): Promise<FilterDefaultOption[]> => {
  const data = await handleApi(`/user/filter-defaults-time/${userid}`)
  return data
}

interface Department {
  department_id: number
}
interface SubjectProperty {
  property_id: number
}

export const getDefaultSearchDepartments = async (
  form: UserId
): Promise<FilterDefaultOption[]> => {
  const data = await handleApi('/user/departments', form)
  return data
}

export const setDefaultSearchDepartment = async (
  form: Department
): Promise<any> => {
  const data = await handleApi('/user/departmentset', form)
  return data
}

export const getSubjectPropertyDefaults = async (
  form: UserId
): Promise<FilterDefaultOption[]> => {
  const data = await handleApi('/user/subjectproperties', form)
  return data
}

export const setSubjectPropertyDefault = async (
  form: SubjectProperty
): Promise<any> => {
  const data = await handleApi('/user/subjectpropertyset', form)
  return data
}
