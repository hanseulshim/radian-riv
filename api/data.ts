export default {
  '/auth/login': {
    token: 'bcf62b2c-c739-42aa-958d-7a9930ca7fff',
    userid_ssid: '12344-3',
    redirect: 'landing-page',
    HasToAcceptTermsOfUse: true,
    HasToAcceptAgentAgreement: true,
    HasToAcceptAppraiserTou: true,
    HasToPassCaptche: true,
    HasToSubmitSecurityQuestions: true,
    HasToCompleteAgentProfile: true,
    NeedsToResetPassword: true
  },
  '/user/get': {
    userid_ssid: '1234567890',
    roleid: 1,
    clientcode: 'Client 1',
    departmentid: 1,
    username: 'user@radian.com',
    email: 'user@radian.com',
    name_last: 'Gunther',
    name_first: 'McDonald',
    address: '82681 Schmidt Plaza',
    city: 'North Carli',
    state: 'WI',
    zip: 58099,
    phone_home: '215-555-1872',
    phone_mobile: '484-555-0980',
    title: 'Manager'
  },
  '/auth/reset': {
    question_text: 'In what year was your father born?',
    question_num: 3
  },
  '/utility/states': [
    {
      label: 'Alabama',
      value: 'AL'
    },
    {
      label: 'Alaska',
      value: 'AK'
    },
    {
      label: 'Arizona',
      value: 'AZ'
    },
    {
      label: 'Arkansas',
      value: 'AR'
    },
    {
      label: 'California',
      value: 'CA'
    },
    {
      label: 'Colorado',
      value: 'CO'
    },
    {
      label: 'Connecticut',
      value: 'CT'
    },
    {
      label: 'Delaware',
      value: 'DE'
    },
    {
      label: 'District of Columbia',
      value: 'DC'
    },
    {
      label: 'Florida',
      value: 'FL'
    },
    {
      label: 'Georgia',
      value: 'GA'
    },
    {
      label: 'Hawaii',
      value: 'HI'
    },
    {
      label: 'Idaho',
      value: 'ID'
    },
    {
      label: 'Illinois',
      value: 'IL'
    },
    {
      label: 'Indiana',
      value: 'IN'
    },
    {
      label: 'Iowa',
      value: 'IA'
    },
    {
      label: 'Kansas',
      value: 'KS'
    },
    {
      label: 'Kentucky',
      value: 'KY'
    },
    {
      label: 'Louisiana',
      value: 'LA'
    },
    {
      label: 'Maine',
      value: 'ME'
    },
    {
      label: 'Maryland',
      value: 'MD'
    },
    {
      label: 'Massachusetts',
      value: 'MA'
    },
    {
      label: 'Michigan',
      value: 'MI'
    },
    {
      label: 'Minnesota',
      value: 'MN'
    },
    {
      label: 'Mississippi',
      value: 'MS'
    },
    {
      label: 'Missouri',
      value: 'MO'
    },
    {
      label: 'Montana',
      value: 'MT'
    },
    {
      label: 'Nebraska',
      value: 'NE'
    },
    {
      label: 'Nevada',
      value: 'NV'
    },
    {
      label: 'New Hampshire',
      value: 'NH'
    },
    {
      label: 'New Jersey',
      value: 'NJ'
    },
    {
      label: 'New Mexico',
      value: 'NM'
    },
    {
      label: 'New York',
      value: 'NY'
    },
    {
      label: 'North Carolina',
      value: 'NC'
    },
    {
      label: 'North Dakota',
      value: 'ND'
    },
    {
      label: 'Ohio',
      value: 'OH'
    },
    {
      label: 'Oklahoma',
      value: 'OK'
    },
    {
      label: 'Oregon',
      value: 'OR'
    },
    {
      label: 'Pennsylvania',
      value: 'PA'
    },
    {
      label: 'Rhode Island',
      value: 'RI'
    },
    {
      label: 'South Carolina',
      value: 'SC'
    },
    {
      label: 'South Dakota',
      value: 'SD'
    },
    {
      label: 'Tennessee',
      value: 'TN'
    },
    {
      label: 'Texas',
      value: 'TX'
    },
    {
      label: 'Utah',
      value: 'UT'
    },
    {
      label: 'Vermont',
      value: 'VT'
    },
    {
      label: 'Virginia',
      value: 'VA'
    },
    {
      label: 'Washington',
      value: 'WA'
    },
    {
      label: 'West Virginia',
      value: 'WV'
    },
    {
      label: 'Wisconsin',
      value: 'WI'
    },
    {
      label: 'Wyoming',
      value: 'WY'
    }
  ],
  counties: [
    {
      label: 'Allegany',
      value: 'allegany'
    },
    {
      label: 'Anne Arundel',
      value: 'anne-arundel'
    },
    {
      label: 'Baltimore',
      value: 'baltimore'
    },
    {
      label: 'Calvert',
      value: 'calvert'
    },
    {
      label: 'Caroline',
      value: 'caroline'
    },
    {
      label: 'Carroll',
      value: 'carroll'
    },
    {
      label: 'Cecil',
      value: 'cecil'
    },
    {
      label: 'Charles',
      value: 'charles'
    },
    {
      label: 'Dorchester',
      value: 'dorchester'
    },
    {
      label: 'Frederick',
      value: 'frederick'
    },
    {
      label: 'Garrett',
      value: 'garrett'
    },
    {
      label: 'Harford',
      value: 'harford'
    },
    {
      label: 'Howard',
      value: 'howard'
    },
    {
      label: 'Kent',
      value: 'kent'
    },
    {
      label: 'Montgomery',
      value: 'montgomery'
    },
    {
      label: "Prince George's",
      value: 'prince-georges'
    },
    {
      label: "Queen Anne's",
      value: 'queen-annes'
    },
    {
      label: 'Somerset',
      value: 'somerset'
    },
    {
      label: "St. Mary's",
      value: 'st-marys'
    },
    {
      label: 'Talbot',
      value: 'talbot'
    },
    {
      label: 'Washington',
      value: 'washington'
    },
    {
      label: 'Wicomico',
      value: 'wicomico'
    },
    {
      label: 'Worcester',
      value: 'worcester'
    }
  ],
  zips: [
    { label: '12345', value: '12345' },
    { label: '23456', value: '23456' },
    { label: '34567', value: '34567' },
    { label: '45678', value: '45678' },
    { label: '56789', value: '56789' }
  ],
  types: [
    { label: 'Single Family Property', value: 'Single Family Property' },
    { label: 'Condo', value: 'Condo' },
    { label: 'Apartment', value: 'Apartment' }
  ],
  msas: [
    { label: 'MSA one', value: 'MSA one' },
    { label: 'MSA two', value: 'MSA two' },
    { label: 'MSA three', value: 'MSA three' }
  ],
  filterDefaults: {
    sqft: {
      value: 1,
      label: 'by Min/Max'
    },
    sqft_min: 12.5,
    sqft_max: 15,
    sqft_percent: {
      value: 2,
      label: '10%'
    },
    comparable_retail: true,
    comparable_distressed: false,
    time_going_back: {
      value: 0,
      label: '3 Months'
    },
    comps_subdivision: true,
    restrict_comps: {
      value: 1,
      label: 'Single Family'
    }
  },
  filterDefaultsSqft: [
    {
      value: 0,
      label: ''
    },
    {
      value: 1,
      label: 'by Min/Max'
    },
    {
      value: 2,
      label: 'by Variance'
    }
  ],
  filterDefaultsSqftPercent: [
    {
      value: 0,
      label: ''
    },
    {
      value: 1,
      label: '5%'
    },
    {
      value: 2,
      label: '10%'
    },
    {
      value: 3,
      label: '15%'
    },
    {
      value: 4,
      label: '20%'
    },
    {
      value: 5,
      label: '30%'
    }
  ],
  filterDefaultsRestrictComps: [
    {
      value: 0,
      label: 'Default'
    },
    {
      value: 1,
      label: 'Single Family'
    },
    {
      value: 2,
      label: 'Multi Family'
    },
    {
      value: 3,
      label: 'Condo'
    },
    {
      value: 4,
      label: 'Mobile/Manufactured'
    }
  ],
  filterDefaultsTime: [
    {
      value: 0,
      label: '3 months'
    },
    {
      value: 1,
      label: '6 months'
    },
    {
      value: 2,
      label: '12 months'
    }
  ],
  departments: [
    {
      value: -1,
      label: '- All -'
    },
    {
      value: -2,
      label: '<Not Assigned>'
    },
    {
      value: 1,
      label: 'Sample 1'
    },
    {
      value: 2,
      label: 'Sample 2'
    },
    {
      value: 3,
      label: 'Sample 3'
    }
  ],
  department: {
    value: -1,
    label: '- All -'
  },
  subjectProperty: {
    value: 0,
    label: 'Include - 6 Months'
  },
  subjectProperties: [
    {
      value: 0,
      label: 'Include - 6 Months'
    },
    {
      value: 1,
      label: 'Include - 3 Months'
    },
    {
      value: 2,
      label: 'Exclude'
    }
  ],
  questions: [
    {
      value: 0,
      label: 'What was your childhood nickname?'
    },
    {
      value: 1,
      label: 'In what city did you meet your spouse/significant other?'
    },
    {
      value: 2,
      label: 'What is the name of your favorite childhood friend?'
    },
    {
      value: 3,
      label: 'What street did you live on in third grade?'
    },
    {
      value: 4,
      label: 'What was the name of your first stuffed animal?'
    },
    {
      value: 5,
      label: 'What is the middle name of your youngest child?'
    }
  ]
}
