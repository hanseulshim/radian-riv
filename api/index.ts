export * from './auth'
export * from './user'
export * from './utility'
import Cookies from 'js-cookie'

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
/* eslint-disable @typescript-eslint/no-explicit-any */
export const handleApi = async (
  route: string,
  payload?: any,
  noToken: boolean = false
) => {
  const myHeaders = new Headers()
  myHeaders.append('Content-Type', 'application/json')
  myHeaders.append('Accept', 'application/json')
  const authCookies = Cookies.get('auth')
  let auth = null
  if (authCookies) {
    auth = JSON.parse(authCookies)
    if (noToken) {
      myHeaders.append('Authorization', `Bearer ${auth.token}`)
    }
  }
  const fetchResponse = payload
    ? await fetch(`${process.env.api}${route}`, {
        method: 'POST',
        headers: myHeaders,
        body: JSON.stringify(payload)
      })
    : await fetch(`${process.env.api}${route}`, {
        method: 'GET',
        headers: myHeaders
      })
  const {
    success,
    status,
    message,
    data
  }: {
    success: boolean
    status: number
    message: string
    data: any
  } = await fetchResponse.json()
  if (!success) {
    throw new Error('Request was not successful')
  } else if (status !== 1) {
    throw new Error(message)
  }
  return data ? data : message
}
