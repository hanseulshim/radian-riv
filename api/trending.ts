import faker from 'faker'

export interface ChartParam {
  range: string
  state: string
  county: string | null
  zip: string | null
  type: string | null
  msa: string | null
}

export const getHomePriceChart = async (params: ChartParam): Promise<any[]> => {
  // const data = await handleApi(`/utility/states/${state}/${county}/zipcodes`)
  const data = []
  for (let i = 0; i < 100; i++) {
    data.push({
      date: faker.date.between('2020-01-01', '2020-12-31'),
      state: faker.random.number(100),
      county: faker.random.number(100),
      zip: faker.random.number(100),
      msa: faker.random.number(100)
    })
  }
  return data
}
