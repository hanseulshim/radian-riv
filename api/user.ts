import Cookies from 'js-cookie'
import { handleApi } from './index'

interface Profile {
  name_first: string
  name_last: string
  title: string
  address: string
  city: string
  state: string
  zip: string
  phone_mobile: string
  phone_home: string
}

export const submitProfile = async (form: Profile): Promise<string> => {
  const data = await handleApi('/user/update', form)
  const authCookies = Cookies.get('auth')
  const { token, user } = JSON.parse(authCookies)
  Cookies.set('auth', { token, user: { ...user, ...form } })
  return data
}

interface FilterDefaultOption {
  label: string
  value: string | number
}

interface FilterDefaults {
  sqft?: FilterDefaultOption
  sqft_min?: number
  sqft_max?: number
  sqft_percent?: FilterDefaultOption
  comparable_retail: boolean
  comparable_distressed: boolean
  time_going_back: FilterDefaultOption
  comps_subdivision: boolean
  restrict_comps: FilterDefaultOption
}

interface FilterDefaultSet {
  sqft?: number
  sqft_min?: number
  sqft_max?: number
  sqft_percent?: number
  comparable_retail: boolean
  comparable_distressed: boolean
  time_going_back: number
  comps_subdivision: boolean
  restrict_comps: number
}

export const getFilterDefaults = async (): Promise<FilterDefaults> => {
  const data = await handleApi(`/user/filter-defaults`)
  return {
    sqft: {
      value: data.sqft.Value,
      label: data.sqft.Label
    },
    sqft_min: data.sqft_min,
    sqft_max: data.sqft_max,
    sqft_percent: data.sqft_percent,
    comparable_retail: data.comparable_retail,
    comparable_distressed: data.comparable_distressed,
    time_going_back: {
      value: data.time_going_back.Value,
      label: data.time_going_back.Label
    },
    comps_subdivision: data.comps_subdivision,
    restrict_comps: {
      value: data.restrict_comps.Value,
      label: data.restrict_comps.Label
    }
  }
}

export const setFilterDefaults = async (
  form: FilterDefaultSet
): Promise<string> => {
  const data = await handleApi('/user/filter-defaults-set', form)
  return data
}

export const getFilterDefaultsSquareFt = async (): Promise<
  FilterDefaultOption[]
> => {
  const data = await handleApi('/user/filter-defaults-sqft')
  return data.sqft.map(option => ({ label: option.Label, value: option.Value }))
}

export const getFilterDefaultsSquareFtPercent = async (): Promise<
  FilterDefaultOption[]
> => {
  const data = await handleApi('/user/filter-defaults-sqft-percent')
  return data.sqft.map(option => ({ label: option.Label, value: option.Value }))
}

export const getFilterDefaultsRestrict = async (): Promise<
  FilterDefaultOption[]
> => {
  const data = await handleApi('/user/filter-defaults-restrict-comps')
  return data.restrictComps.map(option => ({
    label: option.Label,
    value: option.Value
  }))
}

export const getFilterDefaultsTime = async (): Promise<
  FilterDefaultOption[]
> => {
  const data = await handleApi('/user/filter-defaults-time')
  return data.filterTime.map(option => ({
    label: option.Label,
    value: option.Value
  }))
}

export const getSubjectPropertyDefault = async (): Promise<FilterDefaultOption> => {
  const data = await handleApi('/user/subject-property')
  return {
    label: data.Label as string,
    value: data.Value as number
  }
}

export const getSubjectPropertyDefaults = async (): Promise<
  FilterDefaultOption[]
> => {
  const data = await handleApi('/user/subject-properties')
  return data.subjectProperties.map(option => ({
    label: option.Label,
    value: option.Value
  }))
}

export const setSubjectPropertyDefault = async (
  subject_property_id: number
): Promise<any> => {
  const data = await handleApi('/user/subject-properties-set', {
    subject_property_id
  })
  return data
}
