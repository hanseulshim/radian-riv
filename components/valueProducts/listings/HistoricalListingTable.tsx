import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import { IHistoricalListingProperty } from 'api'

interface Props {
  tableData: IHistoricalListingProperty[]
}

export default function HistoricalListingTable({ tableData }: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    setColumns([
      {
        Header: 'Bed',
        accessor: 'bed',
        align: 'right',
        width: 75
      },
      {
        Header: 'Bath',
        accessor: 'bath',
        align: 'right',
        width: 75
      },
      {
        Header: 'SqFt',
        accessor: 'sqft',
        align: 'right',
        width: 75
      },
      {
        Header: 'Garage',
        accessor: 'garage',
        align: 'right',
        width: 75
      },
      {
        Header: 'Lot Size',
        accessor: 'lotSize',
        align: 'right',
        width: 100
      },
      {
        Header: 'Year Built',
        accessor: 'year',
        align: 'right',
        width: 100
      },
      {
        Header: 'Sale Type',
        accessor: 'financeTypeValue',
        align: 'right',
        width: 125
      }
    ])
    setData(tableData)
  }, [tableData])

  return <Table columns={columns} data={data} fetchData={fetchData} />
}
