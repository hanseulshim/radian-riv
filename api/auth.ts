import Cookies from 'js-cookie'
import { handleApi } from './index'
import { squareFootages, timeIntervals, compTypes } from 'utils/constants'

interface Login {
  username: string
  pwd: string
}
export const submitLogin = async (form: Login): Promise<void> => {
  const auth = await handleApi('/auth/login', form)
  Cookies.set('auth', auth)
}

interface ResetPassword {
  username: string
  email: string
}

export const submitResetPassword = async (
  form: ResetPassword
): Promise<string> => {
  const data = await handleApi('/auth/reset', form)
  return data
}

interface Question {
  questionid: number
  question_text: string
}

interface UserQuestion {
  userid_ssid: string
}

export const getUserQuestion = async (
  form: UserQuestion
): Promise<Question> => {
  const data = await handleApi('/auth/question', form)
  return data
}

interface Answer {
  userid_ssid: string
  question_id: number
  answer: string
}

export const submitAnswer = async (form: Answer): Promise<string> => {
  const data = await handleApi('/auth/answer', form)
  return data
}

interface Register {
  name_first: string
  name_last: string
  username: string
  email: string
  phone_mobile: string
}

export const submitRegister = async (form: Register): Promise<string> => {
  const data = await handleApi('/auth/register', form)
  return data
}

interface Profile {
  name_first: string
  name_last: string
  title: string
  address: string
  city: string
  state: string
  zip: string
  department: string
  phone_mobile: string
  phone_home: string
}

export const submitProfile = async (form: Profile): Promise<string> => {
  const data = await handleApi('/user/update', form)
  const authCookies = Cookies.get('auth')
  const auth = JSON.parse(authCookies)
  Cookies.set('auth', { ...auth, user: { ...form } })
  return data
}

interface ChangePassword {
  userid_ssid: string
  pwd: string
}

export const submitChangePassword = async (
  form: ChangePassword
): Promise<string> => {
  const { pwd } = form
  if (pwd.length < 8) {
    throw new Error('Password must be at least 8 characters(s) long')
  } else if (!/[A-Z]/.test(pwd)) {
    throw new Error('Password must contain an uppercase character')
  } else if (!/[a-z]/.test(pwd)) {
    throw new Error('Password must contain a lowercase character')
  } else if (!/[0-9]/.test(pwd)) {
    throw new Error('Password must contain a numeric character')
  } else if (!/[!@#$]/.test(pwd)) {
    throw new Error(
      'Password must contain a special character (example: !,@,#,$)'
    )
  }
  const data = await handleApi('/auth/pwdchange', form)
  return data
}

interface Question {
  questionid: number
  question_text: string
}
export const getSecurityQuestions = async (): Promise<Question[]> => {
  const questions = await handleApi('/auth/questions')
  return questions
}

interface SecurityQuestions {
  userid_ssid: string
  question1id: number
  answer1: string
  question2id: number
  answer2: string
  question3id: number
  answer3: string
}

export const setSecurityQuestions = async (
  form: SecurityQuestions
): Promise<string> => {
  if (
    form.question1id === null ||
    form.question2id === null ||
    form.question3id === null
  ) {
    throw new Error(`Questions can't be blank`)
  }
  if (form.answer1 === 'error') {
    throw new Error('Error with security question')
  }
  const data = await handleApi('/auth/questionsset', form)
  return data
}

interface filterItem {
  label: string
  value: string | number
}

interface FilterDefaults {
  sqFt?: filterItem
  min?: number
  max?: number
  percent?: number
  retail: boolean
  distressed: boolean
  timeGoingBack: filterItem
  onlySubdivisionComps: boolean
  restrictComps: filterItem
}
export const getFilterDefaults = async (): Promise<FilterDefaults> => {
  // const defaults = await handleApi('/auth/getFilterDefaults')
  const defaults = {
    sqFt: {
      label: '1,000 Sq Ft',
      value: 1
    },
    min: null,
    max: null,
    percent: null,
    retail: true,
    distressed: true,
    timeGoingBack: {
      label: '3 months',
      value: 1
    },
    onlySubdivisionComps: false,
    restrictComps: {
      label: 'Single Family',
      value: 2
    }
  }
  return defaults
}

export const setFilterDefaults = async (
  form: FilterDefaults
): Promise<string> => {
  const data = await handleApi('/auth/setFilterDefaults', form)
  return data
}

interface DefaultFilterOptions {
  squareFootages: filterItem[]
  timeIntervals: filterItem[]
  compTypes: filterItem[]
}
export const getFilterDefaultOptions = async (): Promise<DefaultFilterOptions> => {
  // const data = await handleApi('/utility/squarefootages')
  // return data
  const data = {
    squareFootages,
    timeIntervals,
    compTypes
  }
  return data
}
