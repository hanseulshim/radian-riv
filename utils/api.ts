import { AuthType } from 'components/auth/AuthProvider'
import Cookies from 'js-cookie'

//TODO #7 remove once API is set in place
const testAuth = {
  user: {
    userid_ssid: '16252-5',
    roleid: 0,
    clientcode: 'client',
    departmentid: 0,
    username: 'kingalls@boostlabs.com',
    email: 'kingalls@boostlabs.com',
    name_last: 'Ingalls',
    name_middle: 'Nivek',
    name_first: 'Kevin',
    address: '123 Fake Street',
    city: 'Atlanta',
    state: 'GA',
    zip: '55555',
    phone_home: '215-555-1872',
    phone_mobile: '484-555-0980',
    phone_office: '123-555-0980',
    phone_fax: '242-555-0980',
    title: 'AVE User'
  },
  token: 'bcf62b2c-c739-42aa-958d-7a9930ca7fff',
  redirect: 'string',
  status: 0,
  message: 'string'
}

interface Login {
  username: string
  pwd: string
}
export const submitLogin = (form: Login): AuthType => {
  if (form.username !== 'test') {
    throw new Error('Username is not valid')
  } else if (form.pwd !== 'test') {
    throw new Error('Password is not valid')
  }

  return testAuth
}

interface ResetPassword {
  username: string
  email: string
}

export const submitResetPassword = (form: ResetPassword): string => {
  if (form.username === 'error') {
    throw new Error('Error in username')
  } else if (form.email === 'error@boostlabs.com') {
    throw new Error('Error in email')
  }

  return 'Password reset!'
}

interface Register {
  name_first: string
  name_last: string
  username: string
  email: string
  phone_mobile: string
}

export const submitRegister = (form: Register): string => {
  if (form.name_first === 'error') {
    throw new Error('Error in registering user')
  }

  return 'User created!'
}

interface Profile {
  userid_ssid: string
  roleid: number
  clientcode: string
  departmentid: number
  username: string
  email: string
  name_last: string
  name_middle: string
  name_first: string
  address: string
  city: string
  state: string
  zip: string
  phone_home: string
  phone_mobile: string
  phone_office: string
  phone_fax: string
  title: string
}

export const submitProfile = (form: Profile): string => {
  if (form.name_first === 'error') {
    throw new Error('Error in updating user')
  }
  const authCookies = Cookies.get('auth')
  const auth = JSON.parse(authCookies)
  Cookies.set('auth', { ...auth, user: { ...form } })
  return 'Profile updated!'
}
