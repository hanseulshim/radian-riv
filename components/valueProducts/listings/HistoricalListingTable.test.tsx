import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import HistoricalListingTable from './HistoricalListingTable'
import { generateProperties } from '../../../api'
describe('HistoricalListingTable', () => {
  let component
  beforeAll(() => {
    const property = generateProperties(1)
    const props = {
      tableData: property
    }
    component = shallow(<HistoricalListingTable {...props} />)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
