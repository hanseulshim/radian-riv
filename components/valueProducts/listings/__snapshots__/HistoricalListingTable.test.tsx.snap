// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricalListingTable should match snapshot 1`] = `
<Table
  columns={Array []}
  data={Array []}
  fetchData={[Function]}
/>
`;
